app/assets/javascripts/ui_models/app_state/tags_state.ts

```diff
@@ -0,0 +1,53 @@
+import { ContentType, SNSmartTag, SNTag } from '@standardnotes/snjs';
+import { action, computed, makeObservable, observable } from 'mobx';
+import { WebApplication } from '../application';
+
+export class TagsState {
+  tags: SNTag[] = [];
+  smartTags: SNSmartTag[] = [];
+
+  constructor(
+    private application: WebApplication,
+    appEventListeners: (() => void)[]
+  ) {
+    makeObservable(this, {
+      tags: observable,
+      smartTags: observable,
+
+      tagsCount: computed,
+
+      addTagToSelectedNotes: action,
+    });
+
+    appEventListeners.push(
+      this.application.streamItems(
+        [ContentType.Tag, ContentType.SmartTag],
+        async () => {
+          this.tags = this.application.getDisplayableItems(
+            ContentType.Tag
+          ) as SNTag[];
+          this.smartTags = this.application.getSmartTags();
+        }
+      )
+    );
+  }
+
+  async addTagToSelectedNotes(tag: SNTag): Promise<void> {
+    const selectedNotes = Object.values(
+      this.application.getAppState().notes.selectedNotes
```

Comment:
Should this live in the notes state object instead? Looks like a pretty far-reaching call 🤔

Reply:
Yup sounds good, I think I'll move all of these methods to the notes state object